api/include/opentelemetry/trace/trace_state.h

```diff
@@ -45,51 +45,154 @@ class TraceState
   class Entry
   {
   public:
-    Entry() noexcept = default;
-
-    // Copy constructor.
-    Entry(const Entry &copy);
-
-    // Creates an Entry for a given key-value pair.
-    Entry(nostd::string_view key, nostd::string_view value) noexcept;
-
-    nostd::string_view GetKey();
-    nostd::string_view GetValue();
+    Entry() : key_(new char[kKeyMaxSize]), value_(new char[kValueMaxSize]){};
+
+    // Copy constructor
+    Entry(Entry &copy)
+    {
+      key_   = StringToPointer(copy.key_.get(), kKeyMaxSize);
+      value_ = StringToPointer(copy.value_.get(), kValueMaxSize);
+    }
+
+    // Assignment operator
+    Entry &operator=(Entry &other)
+    {
+      key_   = StringToPointer(other.key_.get(), kKeyMaxSize);
+      value_ = StringToPointer(other.value_.get(), kValueMaxSize);
+      return *this;
+    }
+
+    // Create an Entry for a given key-value pair.
+    Entry(nostd::string_view key, nostd::string_view value) noexcept
+    {
+      key_   = StringToPointer(key.data(), kKeyMaxSize);
+      value_ = StringToPointer(value.data(), kValueMaxSize);
+    }
+
+    // Get the key associated with this entry.
+    nostd::string_view GetKey() { return nostd::string_view(key_.get()); }
+
+    // Get the value associated with this entry.
+    nostd::string_view GetValue() { return nostd::string_view(value_.get()); }
+
+    // Set the value for this entry. This overrides the previous value.
+    void SetValue(nostd::string_view value)
+    {
+      value_ = StringToPointer(value.data(), kValueMaxSize);
+    }
 
   private:
     // Store key and value as raw char pointers to avoid using std::string.
-    nostd::unique_ptr<const char[]> key_;
-    nostd::unique_ptr<const char[]> value_;
+    std::unique_ptr<const char[]> key_;
+    std::unique_ptr<const char[]> value_;
+
+    // Copy string into a buffer and return a unique_ptr to the buffer.
+    // This is a workaround for the fact that strcpy doesn't accept a const char* destination.
+    nostd::unique_ptr<const char[]> StringToPointer(const char *str, const int kMaxSize)
+    {
+      nostd::unique_ptr<char[]> temp(new char[kMaxSize]);
+      strcpy(temp.get(), str);
+      return nostd::unique_ptr<const char[]>(temp.release());
+    }
   };
 
   // An empty TraceState.
-  TraceState() noexcept : num_entries_(0) {}
+  TraceState() noexcept : entries_(new Entry[kMaxKeyValuePairs]), num_entries_(0) {}
 
   // Returns false if no such key, otherwise returns true and populates value.
-  bool Get(nostd::string_view key, nostd::string_view value) const noexcept { return false; }
+  bool Get(nostd::string_view key, nostd::string_view value) noexcept
+  {
+    for (int i = 0; i < num_entries_; i++)
+    {
+      if (strcmp(key.data(), entries_[i].GetKey().data()) == 0)
```

Comment:
`nostd::string_view` defines a comparison operator, so `key == entries_[i].GetKey()` should work.